app/controllers/posts_controller.rb

```diff
@@ -310,7 +310,13 @@ def destroy
       RateLimiter.new(current_user, "delete_post_per_day", SiteSetting.max_post_deletions_per_day, 1.day).performed!
     end
 
-    destroyer = PostDestroyer.new(current_user, post, context: params[:context])
+    force_destroy = false
+    if params[:force_destroy].present?
+      guardian.ensure_can_permanently_delete!(post)
```

Comment:
I'm wondering if we have to call `guardian.ensure_can_delete!` above if we're already calling the guardian method here.